app/controllers/services_controller.rb

```diff
@@ -31,4 +33,13 @@ def service_creation_params
       :service_creation
     ).permit(:service_name).merge(current_user: current_user)
   end
+
+  def create_flow
+    if service.flow.blank?
+      ServiceUpdater.new(service.metadata).tap do |service_updater|
+        service_updater.create_flow
+        service_updater.update
+      end
+    end
```

Comment:
Just double checking if I understood. This method is to guarantee that every time that the user enters to edit the service (which is the page flow) we verify if there is no flow and creates one if there isn't. Correct?

Reply:
correctamundo. i made it a before_action coz we want to write a new metadata version to the db before rendering the flow page